spec/factories/factories.rb

```diff
@@ -19,15 +27,18 @@
     skip_create
     smell_detector
     context 'self'
+    source 'dummy_file'
     lines [42]
     message 'smell warning message'
     parameters { {} }
 
     initialize_with do
-      new(smell_detector, context: context,
-                          lines: lines,
-                          message: message,
-                          parameters: parameters)
+      new(smell_detector,
+          source: source,
+          context: context,
+          lines: lines,
+          message: message,
+          parameters: parameters)
```

Comment:
Just a  random note: I really think we should refactor our specs to use factories for all-the-things. 
That would clean up our specs tremendously.
Maybe something to keep in mind..:)